sql/hive-shaded/pom.xml

```diff
@@ -0,0 +1,216 @@
+                  <excludes>
+                    <exclude>META-INF/*.SF</exclude>
+                    <exclude>META-INF/*.DSA</exclude>
+                    <exclude>META-INF/*.RSA</exclude>
+                  </excludes>
+                </filter>
+              </filters>
+              <relocations combine.self="override">
+                <relocation>
+                  <pattern>com.google.common</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.guava</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>com.google.protobuf</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.com.google.protobuf</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>com.google.gson</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.com.google.gson</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>com.github.joshelser</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.com.github.joshelser</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>com.jolbox.bonecp</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.com.jolbox.bonecp</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>com.zaxxer.hikari</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.com.zaxxer.hikari</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>au.com.bytecode</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.au.com.bytecode</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.commons</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.org.apache.commons</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.antlr.runtime</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.org.antlr.runtime</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.velocity</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.org.apache.velocity</shadedPattern>
+                </relocation>
+                <relocation> <!-- from org.antlr:ST4 -->
+                  <pattern>org.stringtemplate.v4</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.org.stringtemplate.v4</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>javolution</pattern>
+                  <shadedPattern>${spark.shade.packageName}.hive.javolution</shadedPattern>
+                </relocation>
+              </relocations>
```

Comment:
`javax.jdo`, `javax.realtime`, `javax.transaction`, ``javax.xml`, `org.json` still exist in the `spark-hive-shaded` jar with the original package name.

Reply:
Oops forgot these. I'm thinking to add these `javax.*` dependencies explicitly in `spark-hive` and exclude them from the uber jar, since they seem relatively stable. What do you think?

On the other hand, seems we should exclude `org.json` from Hive due to SPARK-18262.